synapse/handlers/auth.py

```diff
@@ -727,6 +727,9 @@ def validate_login(self, username, login_submission):
             if canonical_user_id:
                 defer.returnValue((canonical_user_id, None))
 
+        if login_type == LoginType.SSO:
```

Comment:
can you share the thinking behind this change? it looks like it's turning a 400 into a 403, but which endpoint is this for?

Reply:
If I remember correctly, because of this https://github.com/matrix-org/synapse/blob/dc3e58693816b897c612ea2b1d5a9f0656108d7d/synapse/rest/client/v1/login.py#L251-L254
Due to no if condition being met it would fail because SSO is not a valid login type.